clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2720,8 +2723,7 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
   bool handleSyclAccessorType(FieldDecl *FD, QualType FieldTy) final {
     const auto *AccTy =
         cast<ClassTemplateSpecializationDecl>(FieldTy->getAsRecordDecl());
-    if (!AccTy)
-      return false;
+    assert(AccTy && "should not be null.");
```

Comment:
So actually upon reread, I think this is a false-positive.  the 'cast' above should assert on null or wrong type, so this shouldn't be necessary.

Reply:
Thanks @erichkeane for pointing this. I also missed the fact that we are using cast here.